src/document.c

```diff
@@ -3040,6 +3040,24 @@ void document_undo(GeanyDocument *doc)
 					document_undo(doc);
 				break;
 			}
+			case UNDO_INDENT:
+			{
+				undo_action *next_action;
+
+				document_redo_add(doc, UNDO_INDENT, GINT_TO_POINTER(doc->editor->indent_type));
+
+				editor_set_indent(doc->editor, GPOINTER_TO_INT(action->data), doc->editor->indent_width);
+
+				ui_document_show_hide(doc);
```

Comment:
Haven't tested, but I think you need to update the status bar too, and again for redo.